core-bundle/src/Resources/contao/pages/PageLogout.php

```diff
@@ -47,6 +46,15 @@ public function getResponse($objPage)
 			$strRedirect = $objTarget->getAbsoluteUrl();
 		}
 
+		$container = System::getContainer();
+
+		// Redirect immediately if there is no logged in user (see #2388)
+		if (!$container->get('security.helper')->getUser())
```

Comment:
as far as I know, best practice here is to check `$container->get('security.helper')->getToken() instanceof AnonymousToken`. We might also try-catch the logout_url_generator, so any custom handling would work out of the box?

Reply:
> We might also try-catch the logout_url_generator, so any custom handling would work out of the box?

I initially wanted to implement it this way, but since it is only a general `InvalidArgumentException` I thought it might be better to check for a user beforehand instead.